src/lib/y2partitioner/actions/resize_blk_device.rb

```diff
@@ -80,18 +73,49 @@ def resize
       #
       # @return [Boolean] true if the resize action can be performed; false otherwise.
       def validate
-        return true if resize_info.resize_ok?
+        return true if errors.empty?
+
+        # Only first error is shown
+        Yast::Popup.Error(errors.first)
 
-        # TODO: Distinguish the reason why it is not possible to resize, for example:
-        # * partition used by commited LVM or MD RAID
-        # * extended partition with committed logical partitions
+        false
+      end
+
+      # Errors when trying to resize a device
+      #
+      # @return [Array<Strings>]
+      def errors
+        [used_device_error, cannot_be_resized_error].compact
+      end
 
-        Yast::Popup.Error(
-          # TRANSLATORS: an error popup message
-          _("This device cannot be resized.")
+      # Error when trying to resize an used device
+      #
+      # @note A device is being used when it forms part of an LVM or MD RAID.
+      #
+      # @return [String, nil] nil if the device is not being used.
+      def used_device_error
+        return nil unless device.part_of_lvm_or_md?
+
+        # TRANSLATORS: Error message when trying to resize an used device. %{name} is
+        # replaced by the device name (e.g., /dev/sda1).
+        format(
```

Comment:
Same than above. To be sure I would move the comment (I keep only the part about the placeholder).